addons/dynamishelper/dynamishelper.lua

```diff
@@ -24,22 +24,427 @@
 --(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 --SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 --Features
+-- Zone Timer
+-- Time extention tracker
 -- Stagger timer
 -- Currency tracker
 -- Proc identifier
--- Lot currency
+-- Light Luggage Profile
 
 _addon.name = 'DynamisHelper'
-_addon.author = 'Krizz, maintainer: Skyrant'
+_addon.author = 'Krizz, Skyrant'
 _addon.commands = {'DynamisHelper','dh'}
-_addon.version = '1.0.2.0'
+_addon.version = '2.2'
 
-require('strings')
-require('sets')
 config = require('config')
+texts = require('texts')
 res = require('resources')
 
--- Variables
+ProcZones = res.zones:english(string.startswith-{'Dynamis'}):keyset()
+
+-------------------------------------------------------------------------------
+-- Define default values ------------------------------------------------------
+-------------------------------------------------------------------------------
+defaults = T{}
+defaults.text_R = 255
+defaults.text_G = 255
+defaults.text_B = 255
+defaults.Wind_R = 0
+defaults.Wind_G = 255
+defaults.Wind_B = 0
+defaults.Bast_R = 0
+defaults.Bast_G = 0
+defaults.Bast_B = 255
+defaults.Sand_R = 255
+defaults.Sand_G = 0
+defaults.Sand_B = 0
+defaults.pos_x = 0
+defaults.pos_y = 0
+defaults.font_size = 11
+defaults.bg_alpha = 255
+defaults.timer = true
+defaults.tracker = true
+defaults.trposx = 0
+defaults.trposy = 0
+defaults.proc = true
+defaults.pposx = 0
+defaults.pposy = 0
+
+-------------------------------------------------------------------------------
+-- Load defaults from settings.xml --------------------------------------------
+-------------------------------------------------------------------------------
+settings = config.load(defaults)
+
+-- upgrade to new settings file if old settings still exist -------------------
+if settings.trposx and settings.trposy then
+	-- copy position from old config to new one
+   	settings.pos_x = settings.trposx
+   	settings.pos_y = settings.trposy
+	-- delete old settings here ---------------------------------------------------
+	config.save(settings, 'all')
+end
+
+
+windurst_col = "\\cs("..tostring(settings.Wind_R)..","..tostring(settings.Wind_G)..","..tostring(settings.Wind_B)..")"
+bastok_col = "\\cs("..tostring(settings.Bast_R)..","..tostring(settings.Bast_G)..","..tostring(settings.Bast_B)..")"
+sandoria_col = "\\cs("..tostring(settings.Sand_R)..","..tostring(settings.Sand_G)..","..tostring(settings.Sand_B)..")"
+neutral_col = "\\cs(128,128,128)"
+time_col = "\\cs(255,255,0)"
+
+StaggerCount = 0
+current_mob = "unknown"
+current_proc = "unknown"
+currenttime = 0
+obj_time = 0
+end_time = 0
+
+image = texts.new("image")
```

Comment:
I don't know if you actually want this, but if you do, you can make text object changes automatically save to the settings file. So every time you change the position for example, it will automatically save.

To do that, you need to call the constructor like this:
```lua
image = texts.new('image', settings, settings)
```

This will make the settings you currently have obsolete, you'd need to change them to the ones the text library uses. But in return you get many features for free, and the user can completely customize the text box if they want to. You can try it once and you'll see what settings it will generate. If you go that route I'd recommend putting the text settings into its own sub-table:
```lua
defaults = {}
defaults.image = {}

-- and later

image = texts.new('image', settings.image, settings)
```

So the first argument is the settings table for the image object, the second argument is for the root settings object.